bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/discovery/LuxtronikHeatpumpDiscovery.java

```diff
@@ -0,0 +1,231 @@
+    private @Nullable CidrAddress baseIp;
+
+    /**
+     * The {@link ExecutorService} to run the listening threads on.
+     */
+    private @Nullable ExecutorService executorService;
+
+    /**
+     * Constructs the discovery class using the thing IDs that we can discover.
+     */
+    public LuxtronikHeatpumpDiscovery() {
+        super(LuxtronikHeatpumpBindingConstants.SUPPORTED_THING_TYPES_UIDS, FULL_SCAN_TIMEOUT_SECONDS, false);
+    }
+
+    private void setupBaseIp(CidrAddress adr) {
+        byte[] octets = adr.getAddress().getAddress();
+        addressCount = (1 << (32 - adr.getPrefix())) - 2;
+        ipMask = 0xFFFFFFFF << (32 - adr.getPrefix());
+        octets[0] &= ipMask >> 24;
+        octets[1] &= ipMask >> 16;
+        octets[2] &= ipMask >> 8;
+        octets[3] &= ipMask;
+        try {
+            InetAddress iAdr = InetAddress.getByAddress(octets);
+            baseIp = new CidrAddress(iAdr, (short) adr.getPrefix());
+        } catch (UnknownHostException e) {
+            logger.debug("Could not build net ip address.", e);
+        }
+        octet = 0;
+    }
+
+    private synchronized String getNextIPAddress(CidrAddress adr) {
+        octet++;
+        octet &= ~ipMask;
+        byte[] octets = adr.getAddress().getAddress();
+        octets[2] += (octet >> 8);
+        octets[3] += octet;
+        String address = "";
+        try {
+            InetAddress iAdr = null;
+            iAdr = InetAddress.getByAddress(octets);
+            address = iAdr.getHostAddress();
+        } catch (UnknownHostException e) {
+            logger.debug("Could not find next ip address.", e);
+        }
+        return address;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Starts the scan. This discovery will:
+     * <ul>
+     * <li>Request this hosts first IPV4 address.</li>
+     * <li>Send a socket request on port 8888 / 8889 to all IPs on the subnet.</li>
+     * <li>The response is then investigated to see if is an answer from a heat pump</li>
+     * </ul>
+     * The process will continue until all addresses are checked, timeout or {@link #stopScan()} is called.
+     */
+    @Override
+    protected void startScan() {
+        if (executorService != null) {
+            stopScan();
+        }
+
+        CidrAddress localAdr = getLocalIP4Address();
+        if (localAdr == null) {
+            stopScan();
+            return;
+        }
+        setupBaseIp(localAdr);
+        CidrAddress baseAdr = baseIp;
+        scanning = true;
+        ExecutorService localExecutorService = Executors.newFixedThreadPool(SCAN_THREADS);
+        executorService = localExecutorService;
+        for (int i = 0; i < addressCount; i++) {
+
+            localExecutorService.execute(() -> {
+                if (scanning && baseAdr != null) {
+                    String ipAdd = getNextIPAddress(baseAdr);
+
+                    boolean detection = (discoverFromIp(ipAdd, 8889) || discoverFromIp(ipAdd, 8888));
```

Comment:
This var does not seem to be used

Reply:
Yes. The variable isn't needed. But that way it could be written into one line. But can convert it into a if clause instead.